crates/sui-indexer-alt/src/pipeline/concurrent/collector.rs

```diff
@@ -31,8 +31,11 @@ struct Pending<H: Handler> {
 impl<H: Handler> Pending<H> {
     /// Whether there are values left to commit from this indexed checkpoint.
     fn is_empty(&self) -> bool {
-        debug_assert!(self.watermark.batch_rows == 0);
-        self.values.is_empty()
+        let empty = self.values.is_empty();
+        if empty {
+            debug_assert!(self.watermark.batch_rows == 0);
```

Comment:
I think this debug assert is only true when pending is actually empty.

Reply:
Good catch! Makes complete sense.